core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -100,7 +109,16 @@ public static QuerySourceBuilder parseQuerySource(RestRequest request) {
         if (defaultOperator != null) {
             queryBuilder.defaultOperator(Operator.fromString(defaultOperator));
         }
-        return new QuerySourceBuilder().setQuery(queryBuilder);
+        return queryBuilder;
+    }
+
+    public static SearchSourceBuilder getRestSearchSource(BytesReference sourceBytes, IndicesQueriesRegistry queryRegistry)
+            throws IOException {
+        XContentParser parser = XContentFactory.xContent(sourceBytes).createParser(sourceBytes);
+        QueryParseContext queryParseContext = new QueryParseContext(queryRegistry);
```

Comment:
I agree it is trappy. It originally wasn't meant to be created manually I guess, but only through `IndexQueryParserService`. Adding the argument to the constructor might not be enough as the reset would override it anyway.